packages/cluster/src/commands/up.js

```diff
@@ -4,36 +4,29 @@ const { reporter, exec, tryCatchAsync } = require('@dhis2/cli-helpers-engine')
 const {
     initDockerComposeCache,
     makeComposeProject,
-    makeDockerImage,
-    substituteVersion,
+    makeEnvironment,
 } = require('../common')
 
 const defaults = require('../defaults')
 const { seed: doSeed } = require('../db')
 
-const run = async function({
-    name,
-    port,
-    seed,
-    seedFile,
-    update,
-    image,
-    dhis2Version,
-    customContext,
-    ...argv
-}) {
-    const { cluster } = argv
+const run = async function(argv) {
+    const {
+        cluster,
+        name,
+        port,
+        seed,
+        seedFile,
+        update,
+        image,
+        dhis2Version,
+        dbVersion,
+        customContext,
+        variant,
+        channel,
+    } = argv
 
-    const resolvedVersion = dhis2Version || name
-    const resolvedImage = substituteVersion(
-        image || cluster.image || defaults.image,
-        resolvedVersion
-    )
-
-    const resolvedPort = port || cluster.port || defaults.port
-    const resolvedContext =
-        customContext || cluster.customContext || defaults.customContext
-    const resolvedContextPath = resolvedContext ? `/${name}` : ''
+    const runtime = makeEnvironment(argv, {}, cluster)
```

Comment:
The second empty object is going to be replaced with the per-cluster cached configuration. The `cluster` object is the default configuration from `~/.config/d2/config.js`.